src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -51,6 +51,36 @@ public static IResourceBuilder<ParameterResource> AddParameter(this IDistributed
         return builder.AddParameter(name, () => value, publishValueAsDefault, secret);
     }
 
+    /// <summary>
+    /// Creates a new <see cref="ParameterResource"/> that has a generated value using the <paramref name="value"/>.
+    /// </summary>
+    /// <remarks>
+    /// The value will be saved to the app host project's user secrets store when <see cref="DistributedApplicationExecutionContext.IsRunMode"/> is <c>true</c>
+    /// and the lifetime of the resource is <see cref="ContainerLifetime.Persistent"/>.
+    /// </remarks>
+    /// <param name="builder">Distributed application builder</param>
+    /// <param name="name">Name of the parameter</param>
+    /// <param name="value"></param>
+    /// <returns>The created <see cref="ParameterResource"/>.</returns>
+    public static ParameterResource AddPersistentParameter(this IDistributedApplicationBuilder builder, string name, string value)
```

Comment:
What's the difference between this API and `AddParameter(..., persist: true)`?

Reply:
In the end it does the same thing, but you need to use a `ParameterDefault` in the existing one and the one we need is private. So it's either making `ConstantParameterDefault` public or create a new helper method that uses it (which is what I implemented).

Reply:
Can this one be built on top of the other one to show that it is the same thing only with different inputs?

Reply:
Given your current build errors, it kind of shows why this API isn't working very well. You want to pass a ParameterDefault that will generate a temp file when the parameter value isn't already persisted. Basically a callback func that produces the default value when the value isn't already there.